pkg/reconciler/build/build.go

```diff
@@ -158,6 +162,22 @@ func conditionForPod(pod *corev1.Pod) corev1alpha1.Conditions {
 				LastTransitionTime: corev1alpha1.VolatileTime{Inner: metav1.Now()},
 			},
 		}
+	case corev1.PodPending:
+		for _, c := range pod.Status.InitContainerStatuses {
+			if c.State.Waiting != nil &&
+				(c.State.Waiting.Reason == errImagePullMsg || c.State.Waiting.Reason == imagePullBackOffMsg) {
```

Comment:
Should we just always bubble up reason from a waiting container?

Reply:
@matthewmcnew which container's reason? the pod's reason?